rust/src/mime/detect.rs

```diff
@@ -41,3 +42,68 @@ pub unsafe extern "C" fn SCDetectMimeEmailGetData(
 
     return 0;
 }
+
+#[no_mangle]
+pub unsafe extern "C" fn SCDetectMimeEmailGetMultiData(
+    ctx: &MimeStateSMTP, buffer: *mut *const u8, buffer_len: *mut u32,
+    hname: *const std::os::raw::c_char, idx: u32,
+) -> u8 {
+    let c_str = CStr::from_ptr(hname); // unsafe
+    let str = c_str.to_str().unwrap_or("");
+
+    for h in &ctx.headers[..ctx.main_headers_nb] {
+        if mime::slice_equals_lowercase(&h.name, str.as_bytes()) {
+            let mut start = 0;
+            let mut state = FieldCommaState::Start;
+            let mut current_idx = 0;
+
+            for i in 0..h.value.len() {
+                match state {
+                    FieldCommaState::Start => {
```

Comment:
Am I reading this correctly if I think this actually parses a list of email addresses from the `cc` field and returns the addresses one by one? I don't think that is what I'd expect here, I'd expect the rule keyword to simply inspect the contents of the "cc" buffer, w/o further splitting out or parsing.

I guess I could see some value in the logic implemented here as well though, but perhaps only as some dedicated keyword.

Reply:
Oh, okay! So the keyword won’t exactly match the EVE field email.cc[], which matches one address at a time? In this case, it's not a multi-buffer, right? @victorjulien

Reply:
Correct